### ISSUE TITLE
Copying references

### ISSUE THREAD

**Issue by tethal** [qorelanguage/qore#496]
```
%new-style
%enable-all-warnings
%require-types
%strict-args

string s = "A";
reference r1 = \s;
reference r2 = r1;

r1 += "B";
r2 += "C";

printf("%N\n", s);
```

Prints `AB` instead of `ABC`.

One has to use 

```
reference r2 = \r1;
```

to copy references.

**Comment by pavelkveton** [comment]
could it possibly be related to #176?

I believe we have been discussing this already for many times:

string s;
reference r1 = \s;
reference r2 = r1; # IMHO, r2 should contain reference to s

string s;
reference r1 = \s;
reference r2 = \r1; # IMHO, this should be invalid assignment, unless we 
allow for a reference to reference.

**Comment by tethal** [comment]
Looks like `reference` is treated as `any` - the following assignment (which is a shorter version of what is going on in the original post) should cause an error:

```
string s = "A";
reference r2 = s;
```

**Comment by davidnich** [comment]
this was fixed with issues #1814 and #1815
